draft-schlesinger-cfrg-act.md

```diff
@@ -825,11 +840,61 @@ interoperability. The curve specification is included to prevent cross-curve
 attacks and ensure implementations using different curves cannot accidentally
 interact.
 
+### Sigma Proofs and Fiat-Shamir Transform {#sigma-and-fs}
+
+Proofs of knowledge are based on interactive sigma protocols, which
+are made non-interactive through the Fiat-Shamir transform {{FST}}.
+The concrete construction uses two interfaces:
+one for describing the sigma protocol, and
+another for generating non-interactive zero-knowledge proofs.
+
+A proof of knowledge is constructed for the relation
+$$R = {(x, w) : x \in L, w \in W(x)}$$
+where the language (`L`) is the set of linear relations between the
+statement (`x`) and the witness (`w`) in the set (`W(x)`) of valid
+witnesses for `x`.
+The statement is expressed as linear combinations of scalars and group
+elements, while the witness is represented as a list of scalars.
+The LinearRelation interface, described in {{!I-D.irtf-cfrg-sigma-protocols}}, allows
+constructing a sigma protocol for the relation above.
+
+~~~ aasvg
++--------------------------------------------------+
+| LinearRelation                                   |
++--------------------------------------------------+
+| allocate_scalars(n: int): list[int]              |
+| allocate_elements(n: int): list[int]             |
+| append_equation(lhs: int, rhs: list[(int, int)]) |
+| set_elements(e: list[(int, Group.Element)])      |
++--------------------------------------------------+
+~~~
+
+The NISigmaProtocol interface, described in {{!I-D.irtf-cfrg-sigma-protocols}},
+converts an interactive sigma protocol created with the LinearRelation
+interface into a non-interactive proof generation.
+The NISigmaProtocol requires of an initialization vector (`iv`) that
+uniquely identifies the protocol.
+Once initialized, the Prover can generate proofs of knowledge
+of the witness, while the Verifier can validate these proofs.
+
+~~~ aasvg
++--------------------------------------------------+
+| NISigmaProtocol                                  |
++--------------------------------------------------+
+| init(iv: list[byte], rel: LinearRelation)        |
+| prove(witness: list[Group.Scalar]): list[byte]   |
+| verify(proof: list[byte]): boolean               |
++--------------------------------------------------+
+~~~
+
+The NISigmaProtocol is parametrized with a Codec, which specifies
+how to encode prover messages and verifier challenges.
```

Comment:
This isn't actually specified in the draft as far as I can tell?

Reply:
The codec is specified in the FIat-Shamir spec, eg LinearMapCodec for Sigma protocols in 5.1:
https://datatracker.ietf.org/doc/draft-irtf-cfrg-fiat-shamir/

This was a point of confusion - it seems like the `NISigmaProtocol` needs to be initialized with a Codec, but there's no way to initialize the Codec? Maybe the codec for `NISigmaProtocol` defaults to `LinearMapCodec` - but if so, that is not explicitly stated. I asked Michele about this and can let you know what the intended behavior is. In the meantime I have just ignored the Codec in the `NISigmaProtocol` initialization and use.

Reply:
added a reference to Codec section